### ISSUE TITLE
Severity cannot be changed based on environment once `master` branch

### ISSUE THREAD

**Issue by wJoenn** [imdrasil/jennifer.cr#454]
I've done the following in hope of getting no logs in `test` mode
```crystal
levels = {
  "development": Log::Severity::Debug,
  "production":  Log::Severity::Error,
  "test":        Log::Severity::None,
}

Log.setup("db", levels[ENV["KEMAL_ENV"]], Log::IOBackend.new(formatter: Jennifer::Adapter::DBFormatter))
```

And my `spec_helper.cr` file sets the environment as `test`
```crystal
ENV["KEMAL_ENV"] = "test"

require "spec-kemal"
require "../src/app.cr"

require "../db/migrations/*"
Jennifer::Migration::Runner.migrate

Spec.before_each do
  Jennifer::Adapter.default_adapter.begin_transaction
end

Spec.after_each do
  Jennifer::Adapter.default_adapter.rollback_transaction
end
```

But despite that the migrations' logs are still being outputted (`crs` being an alias for `crystal spec`)
For some reason it's not using the Jennifer formatter though
![image](https://github.com/imdrasil/jennifer.cr/assets/75388869/b28b4d03-76ad-4c5b-8e06-479f9ffda5a6)

I used to do this which would work but `conf.logger` is deprecated on `master`
```crystal
Jennifer::Config.configure do |conf|
  if ENV.has_key?("DATABASE_URL")
    conf.from_uri(ENV["DATABASE_URL"])
  else
    conf.read("config/database.yml", ENV["KEMAL_ENV"])
  end

  levels = {
    "development": Log::Severity::Debug,
    "production":  Log::Severity::Error,
    "test":        Log::Severity::None,
  }

  conf.logger = Log.for("db")
  conf.logger.level = levels[ENV["KEMAL_ENV"]]
  conf.logger.backend = Log::IOBackend.new(formatter: Jennifer::Adapter::DBFormatter)
end
```

**Comment by wJoenn** [comment]
Just noticed you addressed the PR that deprecated this https://github.com/imdrasil/jennifer.cr/pull/447#issuecomment-2005009891
I have to admit I don't understand your answer though, sorry about that 🤔

**Comment by imdrasil** [comment]
For this topic - the reason is that migrations just use `puts` to output information to STDOUT about the process. I'll add `verbose_migrations` config to address this. I wouldn't use default logger because it wouldn't work nicely with selected formatter (imho)

Regarding my comment - I added your setup to this repo and it worked as expected. Here all logs by default are written to the memory in tests and you need to set an env variable to actually see them in logs. And at the end it looks like I slightly misunderstood you.

Is it correct that only this portion of "logs" is still written to the output?

**Comment by wJoenn** [comment]
> Is it correct that only this portion of "logs" is still written to the output?

Yes that seems to be the case

**Comment by imdrasil** [comment]
`verbose_migrations` is already there. I'll ensure it works for all cases and make it is a little bit obvious in the docs how to silence migrations as well

**Comment by imdrasil** [comment]
Feel free to reopen this issue if you still see any logs after merged fix you would not expect

**Comment by wJoenn** [comment]
Can confirm this silence the logs in test.
Thanks for the changes Roman !
```crystal
if ENV["KEMAL_ENV"] == "test"
  Log.setup "db", Log::Severity::None, Log::IOBackend.new

  Jennifer::Config.configure do |conf|
    conf.verbose_migrations = false
  end
else
  Log.setup "db", Log::Severity::Debug, Log::IOBackend.new(formatter: Jennifer::Adapter::DBColorizedFormatter)
end
```
